ui/app/selectors/selectors.js

```diff
@@ -538,3 +485,35 @@ export function getLastConnectedInfo (state) {
 export function getIpfsGateway (state) {
   return state.metamask.ipfsGateway
 }
+
+export function getConnectedDomainsForSelectedAddress (state) {
+  const {
+    domains = {},
+    domainMetadata,
+    selectedAddress,
+  } = state.metamask
+
+  const connectedDomains = []
+
+  forOwn(domains, (value, domain) => {
+    const exposedAccounts = flatten(value.permissions.map(
+      (p) => p.caveats?.find(({ name }) => name === 'exposedAccounts').value || []
+    ))
+    if (!exposedAccounts.includes(selectedAddress)) {
+      return
+    }
+
+    const {
+      name,
+      icon,
+    } = domainMetadata[domain] || {}
+
+    connectedDomains.push({
+      key: domain,
+      name,
+      icon,
```

Comment:
This is missing the `extensionId`, which is also in the domain metadata. This is used in the `connected-sites-list` component

Reply:
Good catch, thanks

Reply:
Updated